custom_components/versatile_thermostat/underlyings.py

```diff
@@ -731,8 +731,8 @@ def cap_sent_value(self, value) -> float:
             self._underlying_climate.min_temp is not None
             and self._underlying_climate is not None
         ):
-            min_val = self._underlying_climate.min_temp
-            max_val = self._underlying_climate.max_temp
+            min_val = self._hass.states.get(self._entity_id).attributes.get("min_temp")
+            max_val = self._hass.states.get(self._entity_id).attributes.get("max_temp")
```

Comment:
Why this change ? Is that not exactly the same result ?

Reply:
It is not exactly the same result. Your original method uses the device's raw values. Using the state from HA uses the auto-converted values from HA. A user does not have the option to set what unit a climate entity is in, so accessing the underlying climate entity's temperatures and temperature ranges will return these values in their raw values (in Celsius even if HA is set to Fahrenheit). We don't want this, since this results in double conversion, and inaccurate values for the cap function you have. This would make no difference for someone that has a climate entity in Celsius and their default unit in Celsius, but fixes the problem for mismatched climate entity units.

Reply:
ok. Thank you for the explaination.
I will merge this PR and create a release.

Many thanks for your time to solve this old issue.